universum/modules/output/html_output.py

```diff
@@ -0,0 +1,91 @@
+import os
+import re
+
+from .base_output import BaseOutput
+
+
+__all__ = [
+    "HtmlOutput"
+]
+
+
+class HtmlOutput(BaseOutput):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.filename = None
+
+    def set_artifact_dir(self, artifacts_dir):
+        self.filename = os.path.join(artifacts_dir, "log.html")
+
+    def open_block(self, num_str, name):
+        self.log(name) # stub
+
+    def close_block(self, num_str, name, status):
+        self.log(name) # stub
+
+    def report_error(self, description):
+        self.log(description) # stub
+
+    def report_skipped(self, message):
+        self.log(message) # stub
+
+    def report_step(self, message, status):
+        self.log(message) # stub
+
+    def change_status(self, message):
+        self.log(message) # stub
+
+    def log_exception(self, line):
+        self.log(line) # stub
+
+    def log_stderr(self, line):
+        self.log(line) # stub
+
+    def log(self, line):
+        if self._is_log_start(line):
+            self._write_html_header()
+        self._write_to_file(line) # stub
+        if self._is_log_end(line):
+            self._write_html_footer()
+
+    def log_external_command(self, command):
+        self.log(command) # stub
+
+    def log_shell_output(self, line):
+        self.log(line) # stub
+
+    @staticmethod
+    def _is_log_start(line):
+        log_start_pattern = r"^Universum \d+\.\d+\.\d+ started execution$"
+        return re.match(log_start_pattern, line)
```

Comment:
In my opinion this machinery can be replaced by explicit call from the main universum.